sapp/pipeline/add_features.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+from typing import List, Optional, Set, Tuple
+
+from . import DictEntries, PipelineStep, Summary
+
+log: logging.Logger = logging.getLogger("sapp")
+
+
+class AddFeatures(PipelineStep[DictEntries, DictEntries]):
+    """Optional pipeline that attaches given features to all issues.
+    If the features list is empty, the input is simply passed without
+    change as the output"""
+
+    def __init__(self, features: Optional[List[str]]) -> None:
+        super().__init__()
+        # pyre-fixme[6]: Expected `typing.Iterable[Variable[_T]]` for `set.__init__`
+        # but got `Optional[List[str]]`
+        self.features: Set[str] = set(features)
+
+    def run(self, input: DictEntries, summary: Summary) -> Tuple[DictEntries, Summary]:
+        if len(self.features) > 0:
+            log.info("Attaching provided features")
+        # pyre-fixme[6]: Expected `typing.Sized` for 1st positional only parameter
+        # to call `len` but got `typing.Iterable[sapp.pipeline.ParseIssueTuple]`
+        for i in range(len(input["issues"])):
```

Comment:
You can avoid all the lint errors and using the internal `_replace` API with something like this:
```suggestion
        for issue in input["issues"]:
            issue.feature.append(self.features.difference(set(issue.feature)))
```

Reply:
I tried that but the element in the list wasn't changing for some reason :/

Reply:
Another way of modification works :)